frame/evm/src/runner/stack.rs

```diff
@@ -221,11 +223,21 @@ where
 		//
 		// EIP-3607: https://eips.ethereum.org/EIPS/eip-3607
 		// Do not allow transactions for which `tx.sender` has any code deployed.
-		if is_transactional && !<AccountCodes<T>>::get(source).is_empty() {
-			return Err(RunnerError {
-				error: Error::<T>::TransactionMustComeFromEOA,
-				weight,
-			});
+		// Exception: Allow transactions from EOAs whose code is a valid delegation indicator (0xef0100 || address).
+		if is_transactional {
+			let source_code = <AccountCodes<T>>::get(source);
+			if !source_code.is_empty() {
+				// Check if code is a valid delegation indicator: 0xef0100 + 20-byte address
+				let is_delegation_indicator =
+					source_code.len() == 23 && source_code[0..3] == EIP7702_DELEGATION_INDICATOR;
```

Comment:
Lets move `23` to a constant to some common place. It is being used in other places

```rs
const ADDRESS_SIZE: usize = 20;
const EIP7702_DELEGATION_CODE_SIZE: usize = EIP7702_DELEGATION_INDICATOR.len() + ADDRESS_SIZE;
```

```suggestion
					source_code.len() == EIP7702_DELEGATION_CODE_SIZE && source_code[0..3] == EIP7702_DELEGATION_INDICATOR;
```

Reply:
Addressed in https://github.com/polkadot-evm/frontier/pull/1727/commits/fc7adaf0edaedb1b0591a802dfd339725b7e1a99